### PR TITLE
Logging and error message improvements

### LINKED ISSUE #1875: Provide more information when Key.verify_signature() fails
**Current behavior**:
If we call `Metadata.verify_delegate()` and one of the keys fails to be verified the message they got is this:
https://github.com/theupdateframework/python-tuf/blob/fc9b42fa5d4e887821189055b9f44fa0f04f0885/tuf/api/metadata.py#L413
No logging is provided about why the key has failed verification.

**Expected behavior**:
Adding a log about the last exception message could be useful to provide context for what exactly happened.
This could be useful to identify if a keyvalue is malformed.

### PR DESCRIPTION
**Pull request by jku**
* Improve logging when signature verification fails for unexpected reasons (like key type or key content that securesystemslib can't read)
* Improve error message when deserialization fails

### CODE REVIEW

File: tuf/api/serialization/json.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def deserialize(self, raw_data: bytes) -> Metadata:
             metadata_obj = Metadata.from_dict(json_dict)
 
         except Exception as e:
-            raise DeserializationError from e
+            raise DeserializationError("Failed to deserialize JSON") from e
```
Comment (lukpueh) [inline tuf/api/serialization/json.py:39]: Would it make sense to include the original error message? E.g.:
```suggestion
            raise DeserializationError(f"Failed to deserialize JSON: {e}") from e
```

Same comment applies below.

Comment (jku) [reply tuf/api/serialization/json.py:39]: The few cases that I tried did not end up very useful: the errors can be really varied and are often not informative without more context. This context is IMO only available with the stack trace and source code.

As an example if you remove "scheme" from a key, you would get this is in client example with your suggestion:
`Failed to deserialize JSON: 'scheme'`
there is context there (we're getting a KeyError while parsing the json dict) but this is not understandable from that message, only from the stack trace...

Comment (jku) [reply tuf/api/serialization/json.py:39]: I suggested a related improvement to the client example in  #1862 (optionally show the stack trace, so we usually get something human readable but can do developer readable output as well).

**Comment by jku** [comment]
@lukpueh ping, do you dislike the error message or are we good to go with this one?

**Review by lukpueh — APPROVED** [review]

**Pull request merged**